other/install_2-1_postgresql.sql

```diff
@@ -1516,14 +1516,25 @@ CREATE SEQUENCE {$db_prefix}smileys_seq;
 CREATE TABLE {$db_prefix}smileys (
 	id_smiley smallint DEFAULT nextval('{$db_prefix}smileys_seq'),
 	code varchar(30) NOT NULL DEFAULT '',
-	filename varchar(48) NOT NULL DEFAULT '',
 	description varchar(80) NOT NULL DEFAULT '',
 	smiley_row smallint NOT NULL DEFAULT '0',
 	smiley_order smallint NOT NULL DEFAULT '0',
 	hidden smallint NOT NULL DEFAULT '0',
 	PRIMARY KEY (id_smiley)
 );
 
+#
+# Table structure for table `smiley_files`
+#
+
+CREATE TABLE {$db_prefix}smiley_files
+(
+		id_smiley SMALLINT NOT NULL DEFAULT '0',
+		smiley_set VARCHAR(48) NOT NULL DEFAULT '',
+		filename VARCHAR(48) NOT NULL DEFAULT '',
+		PRIMARY KEY (id_smiley, smiley_set, filename)
```

Comment:
Based on the structur id_smiley and smiley_set should be pk,
you idea could it be to use a index only scan?

Reply:
If you think that only id_smiley and smiley_set need to be primary keys, I believe you.

Reply:
Well yes only this two are pk.

Sometimes in the rdbms world you try to enable index only scan: https://stackoverflow.com/questions/36533644/analog-of-index-only-scans-from-postgresql-in-mysql
for this you add first the "important" column and after this you add optional column to enable the rdbms to use index only scan.

Reply:
Thanks for that link. Always good to learn something new. 🙂